tests/fused_attention_stablehlo_test.py

```diff
@@ -202,5 +203,58 @@ def test_sdpa(self, batch_size: int, seq_len: int, num_heads: int,
       self.assertArraysAllClose(key_grad_ref, key_grad, rtol=1e-5, atol=1e-5)
       self.assertArraysAllClose(value_grad_ref, value_grad, rtol=1e-5, atol=1e-5)
 
+  @unittest.skipIf(len(jax.local_devices()) < 4, '4 gpus required')
```

Comment:
Can you move this into `setUp` instead? We don't allow calls to `jax.local_devices()` on the top-level internally.

Reply:
I moved it to setup now. Thanks!